test/jdk/valhalla/valuetypes/StaticFactoryTest.java

```diff
@@ -70,18 +83,22 @@ public SimpleValue(SimplePrimitive v) {
     @DataProvider
     static Object[][] classes() {
         return new Object[][]{
-                new Object[] { SimplePrimitive.class, true },
-                new Object[] { SimpleValue.class, false },
+                new Object[] { SimpleIdentity.class, true, false, false },
+                new Object[] { SimplePrimitive.class, false, true, true },
+                new Object[] { SimpleValue.class, false, true, false },
```

Comment:
It'd be good to add  a regular class and interface (no `identity` modifier) and other `interface` test cases as well.